### ISSUE TITLE
TypeError: Cannot read property 'current' of undefined

### ISSUE THREAD

**Issue by champgm** [plasticrake/tplink-smarthome-api#56]
I am seeing this error every ~10 seconds. Possibly unrelated, but I recently added an HS110 to my network. Currently using `tplink-smarthome-api@0.23.0`.

```
"stack": "TypeError: Cannot read property 'current' of undefined
  at normalize (<project>/node_modules/tplink-smarthome-api/lib/shared/emeter.js:28:19)
  at Emeter.set realtime [as realtime] (<project>/node_modules/tplink-smarthome-api/lib/shared/emeter.js:35:5)
  at <project>/node_modules/tplink-smarthome-api/lib/plug/index.js:184:29
  at Generator.next (<anonymous>)
  at step (<project>/node_modules/tplink-smarthome-api/lib/plug/index.js:3:191)
  at <project>/node_modules/tplink-smarthome-api/lib/plug/index.js:3:361
  at <anonymous>"
```

I'm guessing it's happening here: https://github.com/plasticrake/tplink-smarthome-api/blob/master/src/shared/emeter.js#L23-L30

My code to watch for plugs on the network is here:
https://github.com/champgm/home-dashboard/blob/b33938d488095e1044c600da7ab3ab31ff777e71/backend/tplink/TpLinkRouter.ts#L77-L83

The error stops if I comment out my `plug-online` handler, which is here: https://github.com/champgm/home-dashboard/blob/b33938d488095e1044c600da7ab3ab31ff777e71/backend/tplink/TpLinkRouter.ts#L60-L65

And that calls a getter, which calls various `tplink-smarthome-api` functions like `getDevice`, `getSysInfo`, and `getPowerState` and it's here:
https://github.com/champgm/home-dashboard/blob/b33938d488095e1044c600da7ab3ab31ff777e71/backend/tplink/TpLinkRouter.ts#L96-L105

I will probably just throw a try/catch around whatever portion of my code is triggering this error, but I thought I should report it anyway.

**Comment by champgm** [comment]
Got some more info, this is the code that was catching the error (I've since modified it to avoid logging this specific error, since it's so noisy)

```
const bunyanLogger: bunyan = LoggerParent.child({ fileName: `${path.basename(__filename)}` });
process.on('unhandledRejection', (error) => {
  if (error.message !== "Cannot read property 'current' of undefined") {
    bunyanLogger.error(error, 'Unhandled promise rejection caught');
  }
});
```

**Comment by champgm** [comment]
I only have 2 plugs on the network right now, here they are, maybe this will help. They may have a couple of attributes added but, other than that, this is what shows up when I

```
{
  "err_code": 0,
  "sw_ver": "1.1.2 Build 161014 Rel.164426",
  "hw_ver": "1.0",
  "type": "IOT.SMARTPLUGSWITCH",
  "model": "HS100(US)",
  "mac": "50:C7:BF:42:BE:C8",
  "deviceId": "80061EE6514CD55E5BCFCFD6D61C76BF183563E1",
  "hwId": "5EACBE93FB9E32ECBE1F1C2ADE6DDE11",
  "fwId": "6C4ACA62CB1B921060CF24148345CD43",
  "oemId": "37589AA1F5CACDC53E2914B7760127E5",
  "alias": "Deck Lights",
  "dev_name": "Wi-Fi Smart Plug",
  "icon_hash": "",
  "relay_state": 0,
  "on_time": 0,
  "active_mode": "schedule",
  "feature": "TIM",
  "updating": 0,
  "rssi": -80,
  "led_off": 0,
  "latitude": xx.xxxxxx,
  "longitude": xx.xxxxx,
  "id": "x.x.x.x",
  "name": "Deck Lights",
  "state": {
    "on": false
  }
}
```

```

{
  "err_code": 0,
  "sw_ver": "1.1.1 Build 160725 Rel.164033",
  "hw_ver": "1.0",
  "type": "IOT.SMARTPLUGSWITCH",
  "model": "HS110(US)",
  "mac": "B0:4E:26:AE:49:F8",
  "deviceId": "8006F1EF8B4E984830206ECC56F2176619A9FF6A",
  "hwId": "60FF6B258734EA6880E186F8C96DDC61",
  "fwId": "060BFEA28A8CD1E67146EB5B2B599CC8",
  "oemId": "FFF22CFF774A0B89F7624BFC6F50D5DE",
  "alias": "Exercise Equipment",
  "dev_name": "Wi-Fi Smart Plug With Energy Monitoring",
  "icon_hash": "",
  "relay_state": 0,
  "on_time": 0,
  "active_mode": "schedule",
  "feature": "TIM:ENE",
  "updating": 0,
  "rssi": -66,
  "led_off": 0,
  "latitude": xx.xxxxx,
  "longitude": xx.xxxxx,
  "id": "x.x.x.x",
  "name": "Exercise Equipment",
  "state": {
    "on": false
  }
}
```

**Comment by sspain** [comment]
I'd run into the same problem. It's because I've introduced two TPLink devices that lack energy metering, but I'm still polling the devices. Every time the library polls a device without an emeter, it faults. I've added code in two locations to correct for this in a pull request I submitted. (I only corrected the plug code; if this works and is acceptable, the change would need to be repeated in the bulb code as well.)
